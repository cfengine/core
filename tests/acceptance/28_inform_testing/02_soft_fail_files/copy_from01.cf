body file control
{
  inputs => { "$(sys.policy_entry_dirname)/../common_soft_fail.cf.sub" };
}

bundle common testcase
{
  vars:
    "filename" string => "$(this.promise_filename)";
}

bundle agent setup
{
  files:
    "$(G.testroot)/TEST.source"
      depth_search => recurse("inf"),
      copy_from => example("$(this.promise_dirname)/test_files");
}

bundle agent main
{
  files:
    "$(G.testroot)/TEST.destination"
      depth_search => recurse("inf"),
      copy_from => example("$(G.testroot)/TEST.source");
}

body copy_from example(from)
{
  source => "$(from)";
  compare => "digest";
  preserve => "true";
}

