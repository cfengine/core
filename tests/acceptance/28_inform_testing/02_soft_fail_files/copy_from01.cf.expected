    info: files promise '/tmp/TEST.destination' repaired, copy_from => example('/tmp/TEST.source'), depth_search => recurse("inf")
    info: Replaced file '/tmp/TEST.destination/subdir/subdir-file' from 'localhost:/tmp/TEST.source/subdir/subdir-file'
    info: Replaced file '/tmp/TEST.destination/file-perms-644' from 'localhost:/tmp/TEST.source/file-perms-644'
